src/cosmoz-treenode-button-view.ts

```diff
@@ -0,0 +1,303 @@
+	searchPlaceholder?: string;
+	noReset?: boolean;
+	searchGlobalPlaceholder?: string;
+	searchMinLength?: number;
+	nodePath?: string;
+};
+
+type ClearItemSelectionParams = {
+	item: Node;
+	ev: Event;
+};
+
+type ObservedAttributes =
+	| 'button-text-placeholder'
+	| 'dialog-text'
+	| 'search-placeholder'
+	| 'search-global-placeholder'
+	| 'node-path'
+	| 'no-reset'
+	| 'search-min-length';
+
+type ButtonViewDialog = HTMLDialogElement & {
+	fit: () => void;
+};
+
+const CosmozNodeButtonView = ({
+	tree,
+	multiSelection = false,
+	dialogText,
+	buttonTextPlaceholder,
+	searchPlaceholder,
+	noReset = false,
+	searchGlobalPlaceholder,
+	searchMinLength,
+	nodePath,
+}: ButtonViewProps) => {
+	const dialogRef = useRef<ButtonViewDialog | null>(null);
+	const searchInputRef = useRef<HTMLInputElement | null>(null);
+
+	const [highlightedNode, setHighlightedNode] = useProperty<Node | null>(
+		'highlightedNode',
+	);
+	const [opened, setOpened] = useState(false);
+	const [nodesOnNodePath, setNodesOnNodePath] = useState<Node[]>([]);
+	const [selectedNodes, setSelectedNodes] = useState<Node[]>([]);
+	const [buttonLabel, setButtonLabel] = useState(buttonTextPlaceholder);
+
+	useEffect(() => {
+		if (nodePath && tree) {
+			setNodesOnNodePath(getTreePathParts(nodePath, tree));
+		}
+	}, [nodePath, tree]);
+
+	useEffect(() => {
+		if (dialogRef.current) {
+			searchInputRef.current = dialogRef.current
+				.querySelector('cosmoz-treenode-navigator')
+				?.shadowRoot?.querySelector('cosmoz-input') as HTMLInputElement | null;
+		}
+	}, [dialogRef.current]);
+
+	const reset = () => {
+		setNodesOnNodePath([]);
+		setSelectedNodes([]);
+		setHighlightedNode(null);
+	};
+
+	const enableReset = () => {
+		if (noReset) {
+			return false;
+		}
+
+		return !!highlightedNode;
+	};
+
+	const clearItemSelection = ({ item, ev }: ClearItemSelectionParams) => {
+		setSelectedNodes(selectedNodes.filter((node) => node !== item));
+		ev.preventDefault();
+		ev.stopPropagation();
+	};
+
+	const getChipText = (node: Node) => {
+		return node.name;
+	};
+
+	const showSelectedNodes = (
+		multiSelection: boolean,
+		selectedNodesLength: number,
+	) => {
+		return multiSelection && selectedNodesLength > 0;
+	};
+
+	const refit = debounce$(() => {
+		dialogRef.current?.fit?.();
+	}, 50);
+
+	const focusSearch = () => {
+		if (searchInputRef.current) {
+			searchInputRef.current.focus();
+		}
+	};
+
+	const onOpen = () => {
+		dialogRef.current?.showModal();
+		setOpened(true);
+		setTimeout(focusSearch, 0);
+	};
+
+	const onClose = () => {
+		setOpened(false);
+		dialogRef.current?.close();
+	};
+
+	useKeyDown('Escape', onClose);
+
+	useEffect(() => {
+		if (!Array.isArray(nodesOnNodePath) || nodesOnNodePath.length === 0) {
+			setButtonLabel(buttonTextPlaceholder || '');
+			return;
+		}
+
+		setButtonLabel(
+			nodesOnNodePath
+				.filter((n) => n)
+				.map((part) => part[tree.searchProperty])
+				.join(' / '),
+		);
+	}, [nodesOnNodePath]);
```

Comment:
`buttonLabel` can be deduced - one less state.

```suggestion
	const buttonLabel = useMemo(() => {
		if (!Array.isArray(nodesOnNodePath) || nodesOnNodePath.length === 0) {
			return buttonTextPlaceholder ?? '';
		}

		return nodesOnNodePath
			.filter((n) => n)
			.map((part) => part[tree.searchProperty])
			.join(' / ');
	}, [buttonTextPlaceholder, nodesOnNodePath]);
```